### ISSUE TITLE
remove toolbar with trash button when user select any row

### ISSUE THREAD

**Issue by suiteappdev** [gregnb/mui-datatables#783]
How to remove toolbar with trash icon when user has selected checkbox row? i dont want show this toolbar.

can any help me please?

**Comment by patorjk** [comment]
You can use the "customToolbarSelect" method to provide a custom toolbar without the trash icon. I don't think you can override the "X row(s) selected" title or prevent the selected toolbar from appearing though.

**Comment by gabrielliwerant** [comment]
Yes, `customToolbarSelect` will help you with what options show in this toolbar. At this point there isn't a built-in option to remove the toolbar entirely. I'm interested in hearing the use case for not having any toolbar (e.g. how would users then take an action based on their selections?).

In any event, you can forcibly remove the select toolbar by overriding css and setting to display: none or something, but be aware that this will not restore the original table toolbar, if that is your intent.

If you want to restore the original toolbar on selections, you can see https://github.com/gregnb/mui-datatables/issues/712 and https://github.com/gregnb/mui-datatables/issues/451 for discussion around this, as it has come up a few times. I'm still not sure it's worth adding that option due to UX issues, but I suggest a path forward to allow easier select toolbar customization, it's just not a priority at the moment.

**Comment by sankho** [comment]
I'm on a project rn working with a design team that doesn't want this feature and is not using any filters beyond basic checkbox selection so there's no worries UX wise -- it would be useful to have a direct approach to hiding things as the ultimate UX can be beyond the ideal definition of the codebase as is IMO.

**Comment by gabrielliwerant** [comment]
@sankho Hello, can you explain a bit more about your use case and user flow?

**Comment by mattspahr** [comment]
> You can use the "customToolbarSelect" method to provide a custom toolbar without the trash icon. I don't think you can override the "X row(s) selected" title or prevent the selected toolbar from appearing though.


Have you found a way to customize the "X row(s) selected"?  I was able to change the "row(s) selected" but cannot get rid of the X . It seems to get it from the length of selectedRows.

This is causing me a lot of trouble. Thanks!

**Comment by raphael10-collab** [comment]
@gabrielliwerant @patorjk How can we keep the toolbar with the trash icon and customize it (adding other options), or, how can we use a customToolbarSelect adding the pre-defined trash icon with its pre-defined functionality?

**Comment by Xaxis** [comment]
It seems odd to me that the "row(s) selected" text remains even after overriding with customToolbarSelect. In my mind the "toolbar" implies that entire div. It's trivial enough to just hide it with CSS, but I'm curious, what's the thinking of leaving this text even when overriding with customToolbarSelect?
